test/lib/PlacesTestUtils.js

```diff
@@ -69,7 +118,28 @@ const PlacesTestUtils = Object.freeze({
         }
       );
     });
-    return (yield promise);
+
+    yield notifPromise;
+    yield insertPromise;
+
+    // poll every 10ms until history items appear on disk
+    yield waitUntil(function*() {
+      let rows = yield getPlacesRows();
+      if ((rows.length - initialNumRows) === urlList.length) {
+        let rowChecked = 0;
+        for (let row of rows) {
+          if (urlSet.has(row.url) && row.frecency > -1 && row.title) {
+            // check if all have completed
+            rowChecked++;
+          }
+        }
+        if (rowChecked === urlList.legnth) {
```

Comment:
i'm currently using this PR as a means to push jobs on travis. sorry about the bug.

I haven't solved the concurrency issue yet, so this is still a WIP, not review-ready yet as i thought unfortunately.